rmcreyes/cpen391-user-app
# Added Basic Layout for Dashboard

# :ship: Pull Request

## :question: Purpose

Added basic layout and functionality for dashboard pages: Parking history, registered cars, payment, current parking status
- Connected with `user/me` backend APIs, supports updating personal information. 
- Other functions (ex: add/edit cars) not wired up yet. 

<p float="left">
  <img src="https://user-images.githubusercontent.com/42983386/108819258-7afaab80-756f-11eb-8855-802ecbc96352.png" width="150" />
  <img src="https://user-images.githubusercontent.com/42983386/108819253-79c97e80-756f-11eb-92f7-94e9a8671478.png" width="150" /> 
 <img src="https://user-images.githubusercontent.com/42983386/108819261-7afaab80-756f-11eb-88fe-f2a1122bd4f3.png" width="150" />
 <img src="https://user-images.githubusercontent.com/42983386/108819262-7afaab80-756f-11eb-921e-0a0653ab4b92.png" width="150" />
  <img src="https://user-images.githubusercontent.com/42983386/108819254-79c97e80-756f-11eb-93f5-f6c093fab581.png" width="150" />
<img src="https://user-images.githubusercontent.com/42983386/108819255-7a621500-756f-11eb-98c8-1a57c44afc11.png" width="150" />
<img src="https://user-images.githubusercontent.com/42983386/108819257-7a621500-756f-11eb-8498-e2773bcd4786.png" width="150" />
<img src="https://user-images.githubusercontent.com/42983386/108819263-7b934200-756f-11eb-9fd2-ccfe3963d86f.png" width="150" />
</p>


## :hammer: Validation Strategy
Manual UI test: 
- After logging in, the user should be navigated to the home page. Ensure that the welcome message includes the user's registered first name. Currently, parked cars should be displayed, if no cars are currently parked, should show the "empty" message
- Clicking on the account icon on the top right of the home page should navigate the user to the accounts page. The accounts page should display the user's registered first and last name, as well as the email. 
   - User can update their basic info on the personal information page, should be alerted that the changes successfully saved after pressing on the "save" button. When navigating back to the account and home pages, ensure that the displayed info is updated. 
- Navigate to the cars page using the bottom navigation. Should currently just display 3 examples of registered cars as a horizontal recycler list. 
   - when clicking on each card view, the user will be navigated to the edit the car information
   - not connected to the backend yet, so will currently not update any information
 - Navigate to the History page using the bottom navigation. Should currently just display 10 identical examples of basic parking history as a vertical list
    - when clicking on each card view, the user will be navigated to the receipt page, which displays more detailed info of the parking history


## :tickets: Ticket(s)

Closes #7
